addons/l10n_latam_check/tests/test_third_party_checks.py

```diff
@@ -88,61 +88,10 @@ def test_02_third_party_check_delivery(self):
         with self.assertRaisesRegex(ValidationError, "Some checks are not anymore in journal,"), self.cr.savepoint():
             self.env['account.payment'].create(vals).action_post()
 
-        operations = self.env['account.payment'].search([('l10n_latam_move_check_ids', '=', check.id), ('state', '=', 'posted')], order="date desc, id desc")
+        operations = self.env['account.payment'].search([('l10n_latam_move_check_ids', '=', check.id), ('state', '!=', 'draft')], order="date desc, id desc")
         self.assertEqual(len(operations), 3, 'There should be 3 operations on the check')
         self.assertEqual(operations, customer_return | supplier_return | delivery)
 
-    def test_03_check_deposit(self):
```

Comment:
Hello ! Valentino told me to ping you on the pr so that we can that case here. Since we have removed the internal_transfer and destination_journal from the payment, this test is failing cause the validation error are not thrown as they should. 
Here in the _get_blocking_l10n_latam_warning_msg function. Do you have any idea on how we could reintroduce those warning checks ?
![image](https://github.com/user-attachments/assets/03245b51-b6e6-4dbe-9d7d-494b70f56610)


For the check on hands, we think that we could check if the payment_method is the in_third_party_checks like the filter for third party checks 
But for the first one we are a bit stuck :thinking: 
Thanks in advance 
@maq-adhoc @jjscarafia

Reply:
@malb-odoo 
I have several comments.
- The payments created from the transfer wizard have no partner_id. Previously I think the transfers recorded the company's partner. Why is this so? 

I made several changes to be able to reject checks I leave a commit in our repo so you can analyze it more easily:

1) I added out_third_party_checks as an outbound method in the bank. 
(https://github.com/adhoc-dev/odoo/commit/85203970de568b284af6fd08294b89e4a729f02f#diff-5ae33b8a1428bc1d1d9fcda20db64a49bf220c2750a52af64c88a392531e8367R12, https://github.com/adhoc-dev/odoo/commit/85203970de568b284af6fd08294b89e4a729f02f#diff-0a3733dca9766e38b72d31811e08ba6100e4ca298f1bc15d785d4d373e8a4fb2R12
2) I modify the wizard to not validate the available journal methods or add the out_third_party_checks method (https://github.com/adhoc-dev/odoo/commit/85203970de568b284af6fd08294b89e4a729f02f#diff-e0b5e82bf72a852d918d7c3a45b50565f9beff4647e419e56b0247365c84675eR43-R47)
3) in payment I add a helper method to know if the payment is a check transfer and use it in the warning https://github.com/adhoc-dev/odoo/commit/85203970de568b284af6fd08294b89e4a729f02f#diff-7e265619580e8fd51cdf8864db73c214fdfe00aed087d8e3ff14b3712642a9d4R49-R54
4) the tests still do not execute all I move the checks from a wizard instead of creating a payment.

With these changes I was able to perform all the flows. I don't know if @vbe-odoo  sees a problem in them.

Reply:
@maq-adhoc thank you for the comments and the commit in your repo.
The only comment is that we could do a `out_third_party_checks` in a Bank journal to cover the use case of a rejected third party check after the user deposit it in the bank, so they can receive it again with a "Check Transfer" from Bank to the "Rejected Third Party Check" journal.

It would be great if we don't allow to pay with a Third Party Check that is in the Bank journal (as the outgoing payment method will exist to cover the return of a rejected 3rd party check), but only allow the "Check Transfer" from the wizard, to return a rejected check. This is because the user is not supposed to pay with a 3rd party check that is in the Bank Journal, as that mean it was deposit and is no longer on hand.

Reply:
@vbe-odoo   I can think of two solutions:
a) Add a constraint that does not allow outgoing third party check payments in bank journals if the partner is defined or is not the company's partner.
b) Create a new retrun method “Rejected Third Party Check” or “Return Third Party Check” available in the bank journals. Block payments when the expected conditions are not met. 

Option b is easier to understand for the user, but requires more changes in the module, more R&D and more test

Reply:
@maq-adhoc thanks for the options!
Let's aim for option "b", if that is too difficult, we can try option "a" instead. So we look for the most user-friendly solution first.
Let me know your thoughts @malb-odoo 

Thanks!

Reply:
@vbe-odoo @malb-odoo 
A few more details that we discussed with @jjscarafia .
It would be better if destination_account_id is not forced in the wizard but computed in the _compute_destination_account_id method. Using the transfer account on payments that use a check method and your partner is not established or is the company's partner.

Payments that contain withholdings should not be posted if the method does not have an account. The withholdings are represented in the journal entry and would be lost. This could happen if a user deletes the oustanding accounts from the journal.

We believe that the 3rd party check flow could work without journal entry. Not so the own checks one.  I don't know if we want it to

Reply:
@maq-adhoc @vbe-odoo
Thanks for those comments :smile: 
Few questions / interrogations: 
"The payments created from the transfer wizard have no partner_id. Previously I think the transfers recorded the company's partner. Why is this so?" 
- Just tested in 17.4 and the two payments created with the check transfer don't have any partner_id, we could get the info from the checks, i've made a suggestion for that, tell me what you think :smile: 

"I modify the wizard to not validate the available journal methods or add the out_third_party_checks method" 
- Don't really get why ? it will remove the check to know if we have the check on hand ?

About the tests that was removed could you do a video or a clear spec on how a rejection should work ? I'm having trouble getting how the thing should behave. Same for claim/return to the customer ? So that i can rework the test and make it work :smile:

Also for: 
"b) Create a new return method “Rejected Third Party Check” or “Return Third Party Check” available in the bank journals. Block payments when the expected conditions are not met."

Do you mean adding a new payment method line on the outbound of the bank ? And what should be the condition to block the payment ? if a payment use that payment method with a bank journal and either a partner is defined or is not the company's partner ?